### DISCUSSION TITLE
csv upload fails

### DISCUSSION THREAD

**Discussion by fmeies** [open-webui/open-webui#6193] [General]
# Bug Report

## Installation Method

Docker

## Environment

- **Open WebUI Version:** 0.3.32
- **Ollama (if applicable):** 0.3.13

- **Operating System:** Ubuntu 22.04.1
- **Browser (if applicable):** Chrome 129

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Uploading csv shoud work.

## Actual Behavior:

Upload failes.

## Description

**Bug Summary:**
I have a test csv consisting of 1000 identical lines with the string "Hello World". Uploading this file results in exactly 2000 POST calls to "/api/embeddings" to my dedicated Ollama instance. These 2000 POST calls take about 250 seconds, however the browser receives a net::ERR_CONNECTION_CLOSED after ~90 seconds. The ui still shows the spinning upload indicated but after a reload of the page the csv file is not listed.

## Reproduction Details

**Steps to Reproduce:**
- Create a simple csv files with 1000 lines
- Upload file via web frontend

## Logs and Screenshots

**Browser Console Logs:**
Collection.svelte:102 File {name: 'upload.csv', lastModified: 1728990283371, lastModifiedDate: Tue Oct 15 2024 13:04:43 GMT+0200 (Mitteleuropäische Sommerzeit), webkitRelativePath: '', size: 14998, …}
index.ts:8         
       POST https://xxx/api/v1/files/ net::ERR_CONNECTION_CLOSED
window.fetch @ fetcher.js:76
c @ index.ts:8
S @ Collection.svelte:132
C @ Collection.svelte:523
Show 1 more frame
Show less
index.ts:22 TypeError: Failed to fetch
    at window.fetch (fetcher.js:76:10)
    at c (index.ts:8:20)
    at S (Collection.svelte:132:31)
    at HTMLInputElement.C (Collection.svelte:523:5)

**Docker Container Logs:**

Not much in the log file of the Open WebUI container:

Tue, Oct 15 2024  1:28:39 pm Collection file-8be73250-5631-4ecb-9b69-f6582202c89e does not exist.
Tue, Oct 15 2024  1:29:29 pm Collection file-de11fd82-0f2b-4e0b-a2ea-aad42ac53814 does not exist.

However, it is somewhat surprising, that Open WebUI seems to duplicate the uploaded file internally. 

This is from the Ollama container:

[GIN] 2024/10/15 - 11:28:35 | 200 |      43.559µs |      10.1.2.180 | GET      "/"
[GIN] 2024/10/15 - 11:28:39 | 200 |    94.79258ms |     10.42.0.118 | POST     "/api/embeddings"
[GIN] 2024/10/15 - 11:28:39 | 200 |   49.254675ms |     10.42.0.118 | POST     "/api/embeddings"
... (~2000 identical lines)
[GIN] 2024/10/15 - 11:31:11 | 200 |   92.063972ms |     10.42.0.118 | POST     "/api/embeddings"
[GIN] 2024/10/15 - 11:31:11 | 200 |   49.321944ms |     10.42.0.118 | POST     "/api/embeddings"
[GIN] 2024/10/15 - 11:31:15 | 200 |      38.019µs |      10.1.2.180 | GET      "/"

We can see that the calls to "/api/embeddings" start the same time as the first log message "Collection file-8be73250-5631-4ecb-9b69-f6582202c89e does not exist" from the Open WebUI container. The browser error message net::ERR_CONNECTION_CLOSE shows up at about 11:28:39 + 90 seconds.

## Additional Information

* Content Extraction Engine: default
* Max Upload Count and Max Upload Count are not set
* Chunk Size: 800, Chunk Overlap: 100

**Comment by nyabutid** [comment]
CSV Uploads do not complete. PDF uploads are successful.

**Comment by bandlaw12** [comment]
Having this issue too - all files (HTML, CSV, XLS, PDF) uploading just fine and landing on server, but OpenWebUI gives same error:

2025/02/16 22:46:12	stdout	Collection file-87a###### does not exist.
2025/02/16 22:46:12	stdout	INFO  [open_webui.routers.retrieval] save_docs_to_vector_db: document <doc name> (1).csv file-87a######
2025/02/16 22:46:02	stdout	INFO  [open_webui.routers.files] file.content_type: text/csv
2025/02/16 22:46:01	stdout	WARNI [python_multipart.multipart] Skipping data after last boundary
